core/embed/prodtest/main.c

```diff
@@ -193,15 +200,22 @@ void extract_params(const char *str, int *numbers, int *count, int max_count) {
 
 static void draw_border(int width, int padding) {
   const int W = width, P = padding, RX = DISPLAY_RESX, RY = DISPLAY_RESY;
-  display_clear();
-  display_bar(P, P, RX - 2 * P, RY - 2 * P, 0xFFFF);
-  display_bar(P + W, P + W, RX - 2 * (P + W), RY - 2 * (P + W), 0x0000);
+
+  gfx_clear();
+
+  gfx_rect_t r_out = gfx_rect_wh(P, P, RX - 2 * P, RY - 2 * P);
+  gfx_rect_t r_in =
+      gfx_rect_wh(P + W, P + W, RX - 2 * (P + W), RY - 2 * (P + W));
+
+  gfx_draw_bar(r_out, COLOR_WHITE);
+  gfx_draw_bar(r_in, COLOR_BLACK);
+
   display_refresh();
 }
 
 static void draw_welcome_screen(void) {
 #if defined TREZOR_MODEL_R || defined TREZOR_MODEL_T3B1
-  display_bar(0, 0, DISPLAY_RESX, DISPLAY_RESY, 0xFFFF);
+  gfx_draw_bar(gfx_rect_wh(0, 0, DISPLAY_RESX, DISPLAY_RESY), 0xFFFF);
```

Comment:
lets use COLOR_WHITE, otherwise it won't work for 32bit colors

Reply:
fixed, see 989fffebf2196106384103bd7800dfdcbb7a54b9